src/marshmallow/fields.py

```diff
@@ -1629,6 +1631,42 @@ def __init__(self, *args, **kwargs):
         self.validators = [validator] + original_validators
 
 
+class IP(String):
+    """A IP address field."""
+
+    default_error_messages = {"invalid_ip": "Not a valid IP address."}
+
+    def _validated(
+        self, value
+    ) -> typing.Optional[typing.Union[ipaddress.IPv4Address, ipaddress.IPv6Address]]:
+        """Format the value or raise a :exc:`ValidationError` if an error occurs."""
+        if value is None:
+            return None
+        if isinstance(value, (ipaddress.IPv4Address, ipaddress.IPv6Address)):
+            return value
+        try:
+            if isinstance(value, (int, bytes)):
+                # ip_address function is flexible in the terms of input value. In the case of
+                # marshalling, integer and binary address representation parsing may lead to
+                # confusion.
+                raise TypeError(
+                    "Only dot-decimal and hexadecimal groups notations are supported. "
+                    "Got %s."
```

Comment:
I can't find a source to back this up but I think we have a consensus on not returning input values in error messages.

Reply:
Fixed.